### ISSUE TITLE
No versions works on OpsWorks (Chef 11)

### ISSUE THREAD

**Issue by pluc** [djoos-cookbooks/newrelic#263]
Went through the issues and saw that people recommended using 2.0.0 (off the Chef supermarket) to install on OpsWorks stacks still using Chef 11.

2.0.0 gives me this error:

`[2016-08-31T14:09:19+00:00] INFO: Forking chef instance to converge...
[2016-08-31T14:09:20+00:00] INFO: *** Chef 11.10.4 ***
[2016-08-31T14:09:20+00:00] INFO: Chef-client pid: 1701
[2016-08-31T14:09:20+00:00] INFO: Setting the run_list to ["opsworks_custom_cookbooks::load", "opsworks_custom_cookbooks::execute"] from JSON
[2016-08-31T14:09:20+00:00] WARN: Run List override has been provided.
[2016-08-31T14:09:20+00:00] WARN: Original Run List: [recipe[opsworks_custom_cookbooks::load], recipe[opsworks_custom_cookbooks::execute]]
[2016-08-31T14:09:20+00:00] WARN: Overridden Run List: [recipe[opsworks_stack_state_sync], recipe[newrelic::default], recipe[newrelic::php_agent], recipe[test_suite], recipe[opsworks_cleanup]]
[2016-08-31T14:09:20+00:00] INFO: Run List is [recipe[opsworks_stack_state_sync], recipe[newrelic::default], recipe[newrelic::php_agent], recipe[test_suite], recipe[opsworks_cleanup]]
[2016-08-31T14:09:20+00:00] INFO: Run List expands to [opsworks_stack_state_sync, newrelic::default, newrelic::php_agent, test_suite, opsworks_cleanup]
[2016-08-31T14:09:20+00:00] INFO: Starting Chef Run for prod02.localdomain
[2016-08-31T14:09:20+00:00] INFO: Running start handlers
[2016-08-31T14:09:20+00:00] INFO: Start handlers complete.
[2016-08-31T14:09:20+00:00] INFO: HTTP Request Returned 404 Not Found: Object not found: /reports/nodes/prod02.localdomain/runs
[2016-08-31T14:09:29+00:00] INFO: Loading cookbooks [apache2, dependencies, deploy, gem_support, mod_php5_apache2, mysql, newrelic, nginx, opsworks_agent_monit, opsworks_aws_flow_ruby, opsworks_bundler, opsworks_cleanup, opsworks_commons, opsworks_initial_setup, opsworks_java, opsworks_nodejs, opsworks_postgresql, opsworks_rubygems, opsworks_stack_state_sync, packages, passenger_apache2, php, rails, ruby, s3_file, scm_helper, ssh_users, test_suite, unicorn]
[2016-08-31T14:09:30+00:00] INFO: newrelic-daemon startup mode: agent
[2016-08-31T14:09:30+00:00] WARN: Cloning resource attributes for service[newrelic-daemon] from prior resource (CHEF-3694)
[2016-08-31T14:09:30+00:00] WARN: Previous service[newrelic-daemon]: /var/lib/aws/opsworks/cache.stage2/cookbooks/newrelic/recipes/php_agent.rb:42:in from_file
[2016-08-31T14:09:30+00:00] WARN: Current  service[newrelic-daemon]: /var/lib/aws/opsworks/cache.stage2/cookbooks/newrelic/recipes/php_agent.rb:105:in from_file
[2016-08-31T14:09:30+00:00] INFO: Running queued delayed notifications before re-raising exception
[2016-08-31T14:09:30+00:00] ERROR: Running exception handlers
[2016-08-31T14:09:30+00:00] ERROR: Exception handlers complete
[2016-08-31T14:09:30+00:00] FATAL: Stacktrace dumped to /var/lib/aws/opsworks/cache.stage2/chef-stacktrace.out
[2016-08-31T14:09:30+00:00] ERROR: resource execute[newrelic-install] is configured to notify resource service[apache2] with action restart, but service[apache2] cannot be found in the resource collection. execute[newrelic-install] is defined in /var/lib/aws/opsworks/cache.stage2/cookbooks/newrelic/recipes/php_agent.rb:29:in from_file
 
[2016-08-31T14:09:30+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)`

Using 2.1.0 gives me this error:

================================================================================
Recipe Compile Error in /var/lib/aws/opsworks/cache.stage2/cookbooks/compat_resource/libraries/autoload.rb
================================================================================
 
 
RuntimeError
------------
This resource is written with Chef 12.5 custom resources, and requires at least Chef 12.0 used with the compat_resource cookbook, it will not work with Chef 11.x clients, and those users must pin their cookbooks to older versions or upgrade.
 
 
Cookbook Trace:
---------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/compat_resource/libraries/autoload.rb:2:in `<top (required)>'
 
 
Relevant File Content:
----------------------
/var/lib/aws/opsworks/cache.stage2/cookbooks/compat_resource/libraries/autoload.rb:
 
1:  unless Gem::Requirement.new(">= 12.0").satisfied_by?(Gem::Version.new(Chef::VERSION))
2>>   raise "This resource is written with Chef 12.5 custom resources, and requires at least Chef 12.0 used with the compat_resource cookbook, it will not work with Chef 11.x clients, and those users must pin their cookbooks to older versions or upgrade."
3:  end
4:  
5:  # If users are on old verisons of ChefDK which activates an (old) gem via cheffish before this cookbook loads, then
6:  # we just try to monkeypatch over the top of a monkeypatch.  Its possible that we have checks in this cookbook which
7:  # will defeat that purpose and fail to monkeypatch on top of monkeypatches -- in which case those checks should be
8:  # removed -- this cookbook needs to win when it gets into a fight with the old gem versions.
9:  if Gem.loaded_specs["compat_resource"]
10:    Chef.log_deprecation "using compat_resource as a gem is deprecated;  please update cheffish and chef-provisioning gems (or use the latest Chef/ChefDK packages) or else manually pin your compat_resource cookbook version to the same version as the gem you are using to remove this warning"
11:  end
 
 
 
[2016-08-31T14:58:38+00:00] ERROR: Running exception handlers
[2016-08-31T14:58:38+00:00] ERROR: Exception handlers complete
[2016-08-31T14:58:38+00:00] FATAL: Stacktrace dumped to /var/lib/aws/opsworks/cache.stage2/chef-stacktrace.out
[2016-08-31T14:58:38+00:00] ERROR: This resource is written with Chef 12.5 custom resources, and requires at least Chef 12.0 used with the compat_resource cookbook, it will not work with Chef 11.x clients, and those users must pin their cookbooks to older versions or upgrade.
[2016-08-31T14:58:38+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)`

Is there still a version that works with Chef 11/OpsWorks?!

Sorry for the bad formatting - I blame GitHub.

**Comment by djoos** [comment]
In #260 Walter (@caffe) reported some luck with 2.0.0 on AWS OpsWorks running Chef 11.10...

@caffe: any chance you could help out @pluc here?

Thanks in advance!
David

**Comment by caffe** [comment]
Hi @pluc,

I did have success using version 2.0.0 of this cookbook. I'm not really specialized is this field and have only basic knowledge. So, I can not fully understand the code you posted but I see it says the resource requires Chef 12. So maybe you haven't tried yet the 2.0.0 version.

I used a Berksfile with the following code:

`source 'https://api.berkshelf.com'
metadata
cookbook 'newrelic', git: 'https://github.com/djoos-cookbooks/newrelic.git', tag: '2.0.0'`

And then I run the following commands on my Stack:

`newrelic::default
newrelic::php_agent`

I followed this article from AWS Forum which helped me a lot:

http://blogs.aws.amazon.com/application-management/post/Tx1T7B0Y37JQER9/Using-New-Relic-to-monitor-applications-on-AWS-OpsWorks

I hope this helps you somehow. :)

Regards,
Walter
